### PR TITLE
Allow proceeding after a blocking condition has been fixed

### PR DESCRIPTION
**Pull request by mvidner**
[bsc#1029291](https://bugzilla.suse.com/show_bug.cgi?id=1029291)

### CODE REVIEW

**Comment by mvidner** [comment]
Tested manually on a CaaSP installation

File: test/widgets_overview_test.rb [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    end
+  end
+
+  describe "#handle" do
+    it "produces a Symbol or nil" do
+      expect(subject.handle(:dummy_event)).to be_a(Symbol).or be_nil
+    end
+  end
+
+  describe "#validate" do
```
Comment (jreidinger) [inline test/widgets_overview_test.rb]: in fact this is wrong, validate does not have to be defined at all. If it is not defined, then widget do not have validation.

File: test/widgets_overview_test.rb [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+# TODO: move these shared examples somewhere to be available to everyone
+RSpec.shared_examples "CWM::AbstractWidget" do
+  describe "#label" do
+    it "produces a String" do
+      expect(subject.label).to be_a String
+    end
+  end
+
+  describe "#handle" do
```
Comment (jreidinger) [inline test/widgets_overview_test.rb]: same here, handle do not have to be defined

Comment (jreidinger) [inline test/widgets_overview_test.rb]: in context of abstract widget, here it is of course defined.

**Pull request merged**